slurm/slurm.py

```diff
@@ -69,6 +69,31 @@ class SlurmTestBase(rfm.RunOnlyRegressionTest):
     num_cpus_per_task = 1
 
 
+@rfm.simple_test
+class Sinfo(SlurmTestBase):
+    descr += ": sinfo partition"
+    executable = "sinfo -s --noheader -o '%P'"
+
+    @run_after('setup')
+    def get_system(self):
+        self.system = rt.runtime().system.name
+
+    @sanity_function
+    def assert_partitions(self):
+        part_map = PARTITION_MAP[self.system]
+        partitions = [x[0] for x in part_map['gpu']] + part_map['smp'] + part_map['mpi']
+        asserts = [sn.assert_found(rf'^{x}$', self.stdout, f'{self.descr} x') for x in partitions]
```

Comment:
did you test this with anansi?

```
✔ [16:59] vsc10042@login1 $VSC_HOME $ sinfo -s --noheader -o '%P' -M anansi
pascal_gpu*
```

The default partition (if any) gets a star

Reply:
does the last x need curly brackets?